### PR TITLE
MINIFICPP-1328 - Improve performance of FileStream

### PR DESCRIPTION
**Pull request by arpadboda**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/src/io/FileStream.cpp [hunk 1]
```diff
@@ -101,13 +96,12 @@ int FileStream::writeData(std::vector<uint8_t> &buf, int buflen) {
...
 int FileStream::writeData(uint8_t *value, int size) {
   if (!IsNullOrEmpty(value)) {
-    std::lock_guard<std::recursive_mutex> lock(file_lock_);
+    std::lock_guard<std::mutex> lock(file_lock_);
     if (file_stream_->write(reinterpret_cast<const char*>(value), size)) {
       offset_ += size;
       if (offset_ > length_) {
         length_ = offset_;
       }
-      file_stream_->seekg(offset_);
```
Comment (arpadboda) [inline libminifi/src/io/FileStream.cpp:110]: Hope I don't miss anything here due to lack of coffee, but it seems to be that there is no real point in seeking to where we just finished writing.

Comment (szaszm) [reply libminifi/src/io/FileStream.cpp:110]: This seeks the input position indicator after write has automatically seeked the output indicator. Yes, they are separate, and no, I don't find this intuitive either.
https://godbolt.org/z/oe1M4j

Comment (fgerlits) [reply libminifi/src/io/FileStream.cpp:110]: I guess the `tellg` and `tellp` pointers could in theory be different, but they have never been, and if the standard library changes this, that will break almost all programs in existence, so I don't think they will. :)

Comment (szaszm) [reply libminifi/src/io/FileStream.cpp:110]: Yup, I assumed that `stringstream` would be similar to `fstream`, but apparently no. No problem in this case, thanks @fgerlits.

**Pull request closed without merging**